src/services/repositories/metadata.ts

```diff
@@ -0,0 +1,86 @@
+import { createConnection, Connection, Repository } from 'typeorm'
+import { SnakeNamingStrategy, } from 'typeorm-naming-strategies'
+
+import { IasqlDatabase, IasqlUser } from '../../metadata/entity/index';
+import * as dbMan from '../db-manager'
+
+class MetadataRepo {
+  private database = 'iasql_metadata';
+  private conn: Connection;
+  private userRepo: Repository<IasqlUser>;
+  private dbRepo: Repository<IasqlDatabase>;
+
+  async init() {
+    const conn = await createConnection(dbMan.baseConnConfig);
+    try {
+      await conn.query(`CREATE DATABASE ${this.database};`);
+    } catch(e) {
+      // postgres doesn't support IF NOT EXISTS for db creation so just ignore the error
```

Comment:
You can avoid this try-catch with a conditional here. I think the following would work:

```plpgsql
DO $$
DECLARE
  db string;
BEGIN
  SELECT datname INTO db
  FROM pg_database
  WHERE datname = ${this.database};
  IF db IS NULL THEN
    CREATE DATABASE ${this.database};
  END IF;
END $$;
```

Reply:
nice

Reply:
actually I had tried a form of this and didn't because `QueryError: CREATE DATABASE cannot be executed from a function` going to leave a comment about that